torchaudio/functional/functional.py

```diff
@@ -864,25 +874,28 @@ def mask_along_axis(
     if mask_param < 1:
         return specgram
 
+    # This line does nothing if specgram is 2D or 3D
     # pack batch
-    shape = specgram.size()
-    specgram = specgram.reshape([-1] + list(shape[-2:]))
+    # shape = specgram.size()
+    # specgram = specgram.reshape([-1] + list(shape[-2:]))
     value = torch.rand(1) * mask_param
     min_value = torch.rand(1) * (specgram.size(axis) - value)
 
     mask_start = (min_value.long()).squeeze()
     mask_end = (min_value.long() + value.long()).squeeze()
     mask = torch.arange(0, specgram.shape[axis], device=specgram.device, dtype=specgram.dtype)
     mask = (mask >= mask_start) & (mask < mask_end)
-    if axis == 1:
+    # unsqueeze the mask if the axis is frequency
+    if axis == dim - 2:
         mask = mask.unsqueeze(-1)
 
     assert mask_end - mask_start < mask_param
 
     specgram = specgram.masked_fill(mask, mask_value)
 
+    # This line does nothing if specgram is 2D or 3D
     # unpack batch
-    specgram = specgram.reshape(shape[:-2] + specgram.shape[-2:])
+    # specgram = specgram.reshape(shape[:-2] + specgram.shape[-2:])
```

Comment:
With a quick skim, I cannot tell if the removal is intended or temporary. Is the intended?

Reply:
it's intended. I commented first just in order to get reviewers' confirmation before I finally remove it. @hwangjeff do you mind taking a look and confirming my judgement? i.e. line 898 and 880 are redundant if specgram is 2D or 3D.

Reply:
they are indeed redundant if `specgram` has 2 or 3 dimensions, but maybe it's worth examining that premise. our spectrogram functions/transforms (which we expect users to use to produce inputs to this function) have the ability to produce outputs of shape `(..., freq, time)`, e.g. https://pytorch.org/audio/stable/transforms.html#torchaudio.transforms.Spectrogram.forward, https://pytorch.org/audio/stable/transforms.html#torchaudio.transforms.MelSpectrogram.forward. to achieve consistency with these, can we generalize the function to work on any tensor of shape `(..., freq, time)`? the line here seems to have been introduced with this in mind

Reply:
good idea thanks